backend/ContextAlgorithm/ContextAlgorithm.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import xml.etree.cElementTree as ET
+import string
+
+INPUT_LANG = 'es'
+OUTPUT_LANG = 'en'
+WORD = 'cama'
+WORD_DIFFICULTY = 882
+TITLE = 'SpanishWordFrequencies'
+PATH = "/home/antonio/Desktop/SE/FrecuencyList/" + TITLE + ".xml"
+MAX_LENGTH_SENTENCE = 15
+
+
+#Class sentence which contains:
+#   -the sentence(self.sentenceString)
+#   -words in the sentence(self.words)
+#   -values of each word(self.wordValues)
+#   -total difficulty of the sentence(self.difficulty)
+class Sentence:
+
+
+    def __init__(self, sentenceString, words):
+        self.sentenceString = sentenceString
+        self.words = words
+        self.wordsValues = []
+        self.difficulty = 0
+
+
+    def addValues(self, values):
+        self.wordsValues = values
+
+
+    def setDifficulty(self, difficulty):
+        self.difficulty = difficulty
+
+
+    def __str__(self):
+        return self.sentenceString
+
+#Delete inneccesary symbols
+def cleanSentence(sentence):
+    sentence = sentence.replace('[...] ', '')
+    sentence = sentence.replace('"', '')
+    sentence = sentence.replace('/', '')
+    return sentence
+
+#Delete the punctuation of a sentence
+def deletePunctuation(sentence):
+    punctuationDeleter = str.maketrans('', '', string.punctuation)
+    sentence = sentence.translate(punctuationDeleter)
+    sentence = sentence.replace('¿', '')
+    sentence = sentence.replace('¡', '')
+    return sentence
+
+#Calculates the error regarding the word difficulty
+def errorCalculation(wordsValues):
+    error=0
+    for x in range(0, len(wordsValues)):
+        error += wordsValues[x] - WORD_DIFFICULTY
+
+    error /= len(wordsValues)
```

Comment:
If you care about decimal points and you are using python version < python 3 then with this expression you are losing precision